step-ca/templates.mdx

```diff
@@ -150,6 +152,95 @@ You can also import values
 from the `"templateData"` object
 in your provisioner's [configuration block](#configuring-the-ca-to-use-templates).
 
+### ASN.1 Values
+
+The X.509 templating system includes functions for ASN.1 encoding in a template.
+This can be useful for populating custom OID `extensions`:
+
+```
+"extensions": [
+  {"id": "1.2.3.4", "critical": true, "value": {{ asn1Enc "int:3848281" | toJson }}}
+]
+```
+
+When applied to template variables, these functions enable dynamic OID values in your certificate:
+
+```
+{
+	"subject": {{ toJson .Subject }},
+	"sans": {{ toJson .SANs }},
+{{- if typeIs "*rsa.PublicKey" .Insecure.CR.PublicKey }}
+	"keyUsage": ["keyEncipherment", "digitalSignature"],
+{{- else }}
+	"keyUsage": ["digitalSignature"],
+{{- end }}
+	"extKeyUsage": ["serverAuth", "clientAuth"],
+	"extensions": [
+		{"id": "1.2.3.5", "value": {{ asn1Marshal .AuthorizationCrt.NotAfter | toJson }}},
+		{"id": "1.2.3.6", "value": {{ asn1Seq (asn1Enc (first .Insecure.CR.DNSNames)) (asn1Enc "int:123456") | toJson }}},
+		{"id": "1.2.3.7", "value": {{ asn1Set (asn1Marshal (first .Insecure.CR.DNSNames) "utf8") (asn1Enc "int:123456") | toJson }}}
+	]
+}
+```
+
+All of the functions here return base64.
+
+To make that possible, `asn1Seq` and `asn1Set` will decode a list of base64 values before putting them into an ASN.1 set or sequence, which is then re-encoded to base64.
+
+For example:
+
+```
+{"id": "1.2.3.4", "value": {{ asn1Seq (asn1Enc "YubiKey") (asn1Enc "int:123456") | toJson }}}
+```
+
+1. `asn1Enc "YubiKey"` will encode "YubiKey" as an ASN.1 Printable string, and return the base64
+2. `asn1Enc "int:123456"` will encode 123 as an ASN.1 Integer, and return the base64
+3. `ans1Seq` wil get all the base64 strings, decode them, create an ASN.1 SEQENCE and return the base64 of the SEQUENCE.
+4. Finally, `toJson` returns a JSON string representation of the base64
+
+#### asn1Enc
+
+Encodes a string into an ASN.1 value.
+A data type can be supplied inline.
```

Comment:
It supports the types described in [this function](https://github.com/smallstep/crypto/blob/878092791d7e5ef8530dfa949cfe3cde61373d77/x509util/extensions.go#L486):
* int: "int:1234"
* oid: "oid:1.2.3.4"
* utf8: "utf8:string"
* ia5: "ia5:srtring"
* numeric: "numeric:1234"
* printable: "printable:string" or just "string"
* utc: "utc:2023-03-29T02:03:57Z" or "utc:2023-04-20 15:28:27.909088 -0700 PDT"
* generalized: "generalized:2023-03-29T02:03:57Z" or "generalized:2023-04-20 15:28:27.909088 -0700 PDT"
* raw: a special one that allows any base64 string